### ISSUE TITLE
rendering of waterway=ditch/drain incentivizes bad mapping

### ISSUE THREAD

**Issue by imagico** [gravitystorm/openstreetmap-carto#2346]
Currently this style renders waterway=ditch/drain thinner than waterway=stream at z15+ (2 vs. 3 pixels) but otherwise identical leading to widespread abuse for tagging small streams - there are literally thousands of cases of this in mountainous parts of southern Germany and Switzerland.

I have no solution for this at the moment, rendering them the same would reduce the incentive but ideally rendering should clearly indicate the difference between natural and artificial waterways.

Related to #1101

**Comment by ray66** [comment]
The width and zoom level visibility should reflect the appearance of natural and artificial waterways in the real world. According the [waterway wiki page](http://wiki.openstreetmap.org/wiki/Key:waterway)  the major difference between streams and ditches is that the latter are artificial. The pictures illustrating both values are very similar. According to the same wiki page, the width of a ditch is similar to that of a small river. Therefore, my proposal is

- render ditches with the same width as streams 
- render ditches and streams less thick than currently and only from zoom level 15 or 16 on
- render drains thicker than drains, slightly less or equal to rivers

**Comment by kocio-pl** [comment]
> render drains thicker than drains

What did you mean here?

**Comment by imagico** [comment]
> The width and zoom level visibility should reflect the appearance of natural and artificial waterways in the real world.

No, a map is always a more or less abstract depiction.  Nearly all maps that display rivers and other waterways display the important ones earlier and more prominently than they would appear in reality.  The ability of this style to do the same is primarily limited by the difficulty to determine a reasonable and consistent measure of importance in real time from local properties.

Practically `waterway=ditch` and `waterway=drain` are used almost interchangeably, in particular since ditch is defined through the purpose of drainage.  There is not really a basis in the data for rendering these two in different ways.

But it would make sense to make a clear difference between artificial and natural waterways in rendering, i am just not sure what would be a good way to do this.

**Comment by ray66** [comment]
@kocio-pl : Sorry, it should read "drains thicker than **ditches**". I corrected my original post.

**Comment by jeisenbe** [comment]
FYI, @imagico wrote a blog post showing a possible option for fixing this issue:

http://blog.imagico.de/water-under-the-bridge/

Complete code available here: https://github.com/imagico/osm-carto-alternative-colors/blob/master/water.mss

```
[waterway = 'ditch'],
      [waterway = 'drain'] {
        [zoom >= 15][intermittent != 'yes'][seasonal != 'yes'] {
          water/line-color: darken(@river-color, 6%);
          artificial/line-color: lighten(@river-color, 6%);
          artificial/line-width: @stream-width-z15 * 0.5;
          [zoom >= 16] { artificial/line-width: @stream-width-z16 * 0.5; }
          [zoom >= 17] { artificial/line-width: @stream-width-z17 * 0.5; }
          [zoom >= 18] { artificial/line-width: @stream-width-z18 * 0.5; }
          artificial/line-cap: butt;
          artificial/line-join: round;
        }
      }
```

This seems like a nice solution: ditches and drains are rendered the same width at streams (3 pixels at z16, 4 pixels at z18) but a lighter centerline is used to distinguish the artificial waterways: ditches, drains (and canals)

**Comment by Adamant36** [comment]
@jeisenbe, you should test it out.

**Comment by imagico** [comment]
Note the fix i had in mind as a good first issue was simple rendering ditches/drains identical to streams.  Adding a subtle difference to stream that does not imply a size difference would be a bonus.

The ac-style approach is obviously connected to the three color scheme for waterbodies - not sure if and how it can be ported here.

**Comment by jeisenbe** [comment]
I'm planning to submit a PR with the darker river color, since this is
needed to get the better low zoom rendering and for better contrast of
rivers on woodlands.

But I think that something like a 12% lighter centerline color (As in
the AC style) would work even if the current ditch/drain color was not
changed.

**Comment by imagico** [comment]
The idea of the 6% darker/lighter was that the overall visual weight of the line is the same as for a natural waterway.  If this also works and is recognizable with a different, lighter color would need to be tested.

**Comment by imagico** [comment]
I analyzed the width distribution of waterways with a width tag [here](https://imagico.de/blog/en/the-ways-of-the-water-in-openstreetmap/).  This confirms that, at least for `waterway=ditch`, there is not a solid basis for rendering them more narrow or later than `waterway=stream` - independent of the argument of mapping incentives.
